packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -661,7 +655,7 @@ export class BasicCrawler<
      * then retries them in a case of an error, etc.
      */
     protected async _runTaskFunction() {
-        const source = this.requestQueue || this.requestList!;
+        const source = this.requestQueue || this.requestList || await this.getRequestQueue();
```

Comment:
Is this to enable crawlers automatically using the default queue?

Reply:
Yop, we can't do this in the constructor as it needs to be awaited. Next time `_runTaskFunction` will be invoked it will end up with `this.requestQueue` being set.